### PR TITLE
Numpy speedups

### PR DESCRIPTION
**Pull request by adampbeardsley**
Should this be merged in?

### CODE REVIEW

**Comment by mkolopanis** [comment]
ah yeah I was working on this, but wanted to get something else in first I remember. Let me get going on this again.

**Comment by mkolopanis** [comment]
Here's the jupyter notebook i used to develop/test these changes. While there are some speed ups which is nice, there's a considerable reduction in the memory bloat in a lot of these functions too.

[numpize_for_loops.pdf](https://github.com/epic-astronomy/LWA_EPIC/files/6983486/numpize_for_loops.pdf)

**Comment by mkolopanis** [comment]
I think at this point we might as well get this reviewed and in/out and then we can come back from another wave of changes later if we need them.

File: LWA_EPIC/LWA_EPIC.py [hunk 1]
```diff
@@ -210,37 +203,20 @@ def GenerateLocations(
...
-    # Tile them for ntime...
-    locx = numpy.tile(lsl_locsf[0, ...], (ntime, 1, 1, 1))
-    locy = numpy.tile(lsl_locsf[1, ...], (ntime, 1, 1, 1))
-    locz = numpy.tile(lsl_locsf[2, ...], (ntime, 1, 1, 1))
-    # .. and then stick them all into one large array
-    locc = numpy.concatenate([[locx, ], [locy, ], [locz, ]]).transpose(0, 1, 3, 4, 2).copy()
+    lsl_locsf += (grid_size - np.max(lsl_locsf, axis=3, keepdims=True)) / 2.
 
+    # add ntime axis
+    locc = np.broadcast_to(lsl_locsf, (ntime, 3, npol, nchan, lsl_locs.shape[1],)).transpose(1, 0, 3, 4, 2).copy()
```
Comment (adampbeardsley) [inline LWA_EPIC/LWA_EPIC.py]:
```suggestion
    locc = np.broadcast_to(lsl_locsf, (ntime, 3, npol, nchan, lsl_locs.shape[1])).transpose(1, 0, 3, 4, 2).copy()
```

**Comment by adampbeardsley** [comment]
Looks great, thanks @mkolopanis . Unfortunately, I can't approve the PR since I'm the one who opened it.

**Review by mkolopanis — APPROVED** [review]
I hereby approve my own work on @adampbeardsley behalf

**Pull request merged**